read-fonts/src/tables/layout/script.rs

```diff
@@ -0,0 +1,269 @@
+            // they're really trying to support Thai, for example :(
+            Tag::new(b"latn"),
+        ] {
+            if let Some(index) = self.index_for_tag(tag) {
+                return Some(SelectedScript {
+                    tag,
+                    index,
+                    is_fallback: true,
+                });
+            }
+        }
+        None
+    }
+}
+
+/// A prioritized list of OpenType script tags mapped from a Unicode script
+/// tag.
+///
+/// Generated by [`script_tags_from_unicode`].
+#[derive(Copy, Clone, PartialEq, Eq, Default)]
+pub struct ScriptTags {
+    tags: [Tag; 3],
+    len: usize,
+}
+
+impl Deref for ScriptTags {
+    type Target = [Tag];
+
+    fn deref(&self) -> &Self::Target {
+        &self.tags[..self.len]
+    }
+}
+
+impl std::fmt::Debug for ScriptTags {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{:?}", self.deref())
+    }
+}
+
+/// Given a [Unicode script code](https://unicode.org/iso15924/iso15924-codes.html),
+/// returns a prioritized list of matching
+/// [OpenType script tags](https://learn.microsoft.com/en-us/typography/opentype/spec/scripttags).
+///
+/// See [hb_ot_all_tags_from_script](https://github.com/harfbuzz/harfbuzz/blob/63d09dbefcf7ad9f794ca96445d37b6d8c3c9124/src/hb-ot-tag.cc#L155C1-L155C27)
+/// for the equivalent HarfBuzz function.
+pub fn script_tags_from_unicode(unicode_script: Tag) -> ScriptTags {
+    let mut tags = [Tag::default(); 3];
+    let mut len = 0;
+    if let Some(new_tag) = new_tag_from_unicode(unicode_script) {
+        // Myanmar maps to mym2 but there is no mym3
+        if new_tag != Tag::new(b"mym2") {
+            let mut bytes = new_tag.to_be_bytes();
+            bytes[3] = b'3';
+            tags[len] = Tag::new(&bytes);
+            len += 1;
+        }
+        tags[len] = new_tag;
+        len += 1;
+    }
+    tags[len] = old_tag_from_unicode(unicode_script);
+    len += 1;
+    ScriptTags { tags, len }
+}
+
+// See <https://github.com/harfbuzz/harfbuzz/blob/63d09dbefcf7ad9f794ca96445d37b6d8c3c9124/src/hb-ot-tag.cc#L37>
+fn old_tag_from_unicode(unicode_script: Tag) -> Tag {
+    let mut bytes = unicode_script.to_be_bytes();
+    let tag_bytes = match &bytes {
+        // Katakana and Hiragana both map to 'kana'
+        b"Hira" => b"kana",
+        // Spaces at the end are preserved, unlike ISO 15924
+        b"Laoo" => b"lao ",
+        b"Yiii" => b"yi  ",
+        // Unicode 5.0 additions
+        b"Nkoo" => b"nko ",
+        // Unicode 5.1 additions
+        b"Vaii" => b"vai ",
+        _ => {
+            // Else, just change the first char to lowercase
+            bytes[0] = bytes[0].to_ascii_lowercase();
+            &bytes
+        }
+    };
+    Tag::new(tag_bytes)
+}
+
+// See <https://github.com/harfbuzz/harfbuzz/blob/63d09dbefcf7ad9f794ca96445d37b6d8c3c9124/src/hb-ot-tag.cc#L84>
+fn new_tag_from_unicode(unicode_script: Tag) -> Option<Tag> {
```

Comment:
I've definitely impl'd this somewhere before... looks like it's in `fontbe`? Let's open an issue to deduplicate.

Reply:
it's in a different repo, at the very least we need to do a release of read-fonts before we can change anything there.

Reply:
Oh sorry, I skimmed pastit being in fontc not here. Then +1 to issue or fast-follow PR to update fontc.

Reply:
@cmyr I exposed a `UNICODE_TO_NEW_OPENTYPE_SCRIPT_TAGS` const that is basically the same as `NEW_SCRIPT_TAGS` in fontbe except the key is `&[u8; 4]` instead of `&str`. Does that work for you?

Reply:
i'm not going to have time to dig into this before I take off, so let's merge as-is and we can figure that out when I get back.